src/slot.js

```diff
@@ -72,18 +72,21 @@ export function Slot(props) {
         const { attributes } = slot;
         for (let i = 0, l = attributes.length; i < l; i++) {
           const { name, value } = attributes[i];
-          if (name == 'name') {
-            // This is the slot's name.
-          } else if (!node.hasAttribute(name)) {
-            // TBD: this means that attributes can be rendered only once?
-            // TBD: what do we do with style and class?
+          if (name == 'name' ||
+              name == 'class' ||
+              name == 'style' ||
+              name == 'i-amphtml-context') {
+            // This is the slot's name or other internal attributes.
+          } else {
+            // TBD: switch to explicit definition of "supported" attributes.
             node.setAttribute(name, value);
           }
         }
         // Boolean attributes:
         node.disabled = slot.hasAttribute('disabled');
         node.hidden = slot.hasAttribute('hidden');
         toggleAttribute(node, 'selected', slot.hasAttribute('selected'));
+        toggleAttribute(node, 'expanded', slot.hasAttribute('expanded'));
```

Comment:
This doesn't seem scalable to me. It seems like every AMP element will have to register what attributes it cares so synchronized between the slot and the element.

Reply:
I definitely agree. I think our `opts` system is definitely a way to go here. I'll do a separate PR to update all these cases. But also important to consider: there's definitely a small set of attributes that we would want to propagate to all slots. At the very least: `hidden` and `disabled`.